src/thunderbots/software/ai/navigator/obstacle/placeholder_obstacle.cpp

```diff
@@ -0,0 +1,21 @@
+#include "ai/navigator/obstacle/placeholder_obstacle.h"
+
+Polygon PlaceholderObstacle::getBoundaryPolygon(const Robot& robot,
+                                                double robotRadiusScaling,
+                                                double velocityProjectionScaling)
+{
+    (void)robot;
+    (void)velocityProjectionScaling;
+    static double ROOT_THREE_BY_TWO =
+        0.86602540378;  // static, so it's not recalculated 4 times per call
+    double radius = ROBOT_MAX_RADIUS_METERS * robotRadiusScaling;
```

Comment:
You could use `Point::createFromAngle`, multiply it by the radius, and add it to the centre point instead of having this magic constant, and that would make it more clear what's actually going on here.

Reply:
I don't think it matters since this is just a placeholder obstacle and is mainly for testing. I will keep this in mind for the velocity obstacle.

Reply:
So this brings to mind an important quote, namely "premature optimization is the root of all evil" (see "When To Optimize" [here](https://en.wikipedia.org/wiki/Program_optimization) ). In general there will be a very small portion of any given piece of software that will make up the majority of the program time. In our case we've got network latency, gradient descent, etc.. So in general we really don't ever need to worry about optimizations like this just because we have bits of our software that are orders of magnitude more expensive.

I do however think that, in general, being aware of this stuff is great!

_(I also totally get this is just for testing, I just thought it was a good opportunity to explain the above point)_